aws-java-sdk-dynamodb/src/test/java/com/amazonaws/services/dynamodbv2/document/utils/ValueMapAndWithJsonSupportTest.java

```diff
@@ -0,0 +1,44 @@
+package com.amazonaws.services.dynamodbv2.document.utils;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.*;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+
+import com.amazonaws.AmazonClientException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Test;
+
+public class ValueMapAndWithJsonSupportTest {
+
+    private static final String NO_JSON_STRING = "nojson";
+    private static final String KEY = "somekey";
+
+    @Test(expected = AmazonClientException.class)
+    public void valueMapCreationShouldFailIfNoJsonStringIsUsedAsValue() {
+        new ValueMap().withJSON("a", NO_JSON_STRING);
+    }
+
+    @Test
+    public void valueMapShouldReturnAProperDeserializedJsonObject() {
```

Comment:
Be cool to see a test for a JSON list too - assume this method should support that also?